examples/misc/raptor.py

```diff
@@ -1,4 +1,29 @@
 #!/usr/bin/env python3
+"""Demonstrate the "raptor" features for remote Task management.
+
+This script and its supporting files may use relative file paths. Run from the
+directory in which you found it.
+
+Refer to the ``raptor.cfg`` file in the same directory for configurable run time details.
+
+By default, this example uses the ``local.localhost`` resource with the ``local``
+access scheme. Before running, confirm that you are able to ssh to localhost
+without being prompted for a password.
+
+In this example, we
+* Launch one or more raptor "master" task(s), which self-submits additional tasks.
+  (TODO: How do we check/confirm their results?)
+* Stage scripts to be used by a raptor "Worker" (TODO: unused?)
+* Provision a Python virtual environment with :py:func:`~radical.pilot.prepare_env`
+* Submit several tasks that will be routed through the master(s) to the worker(s).
+* Submit a non-raptor task in the same Pilot environment
+
+The raptor.cfg file should be adjusted to avoid oversubscribing your resources.
+In particular,
+TODO: what are the constraints of total cores, master cores, worker
+    and task cores with respect to each other? And what are the corresponding
+    TaskDescription fields?
```

Comment:
Up to `cores_per_host` masters can run on a node (`masters_per_node`).  That's the only real constraint I think.  Otherwise the arithmetic in `raptor.py` should guarantee that the pilot has enough resources and that no oversubscription happens.

> And what are the corresponding TaskDescription fields?

The given values ultimately defined the number of cores used by a master task (`threads_per_process`) and number of ranks used for each worker (`cpu_processes`).

Reply:
`cores_per_host` does not appear anywhere in the project. Do you mean `cores_per_node` from the config file?

`threads_per_process` does not occur in the project. Do you mean `td.cpu_threads`?

> Otherwise the arithmetic in raptor.py should guarantee that the pilot has enough resources and that no oversubscription happens.

Do you agree with [the assessment](https://github.com/radical-cybertools/radical.pilot/issues/2218#issuecomment-1239824592) that my most recent tests for #2218 failed due to lack of resources? If so, what additional documentation or checks can you suggest in order to mitigate?